interface/modules/zend_modules/module/Installer/src/Installer/Controller/InstallerController.php

```diff
@@ -84,36 +94,20 @@ public function registerAction()
         $request  = $this->getRequest();
         if ($request->isPost()) {
             if ($request->getPost('mtype') == 'zend') {
+                // TODO: We want to be able to load the modules
+                // from the database.. however, this can be fairly slow so we might want to do some kind of APC caching of the module
+                // list that is loaded using the OpenEMR db connector and not the zend db connector, cache the modules, and then
+                // we can filter / update that list.  We'll have to inject the unloaded module list into the installer but that is fine.
                 $rel_path = "public/".$request->getPost('mod_name')."/";
-                $fName = $GLOBALS['srcdir']."/../".$GLOBALS['baseModDir'].$GLOBALS['zendModDir']."/config/application.config.php";
-                $tmp = include $fName;
-                $modName = trim($request->getPost('mod_name'));
-                $module_exist_in_config = false;
-                if (in_array($modName, $tmp['modules'], true)) {
-                    $module_exist_in_config = true;
-                }
 
+                // registering the table inserts the module record into the database.
+                // it's always loaded regardless, but it inserts it in the database as not activated
                 if ($this -> getInstallerTable() -> register($request->getPost('mod_name'), $rel_path, 0, $GLOBALS['zendModDir'])) {
-                    //add the Module name in the application config file if not already present
-                    $fileName = $GLOBALS['srcdir']."/../".$GLOBALS['baseModDir'].$GLOBALS['zendModDir']."/config/application.config.php";
-                    $data = include  $fileName;
-                    //TODO what if same name is already there for another module
-                    $data['modules'] = array_merge($data['modules'], array($request->getPost('mod_name')));
-                    //recreate the config file
-                    if (is_writable($fileName)) {
-                        if (!$module_exist_in_config) {
-                            $content = "<?php return array(";
-                            $content .= $this->getContent($data);
-                            $content .= ");";
-                            file_put_contents($fileName, $content);
-                        }
-                    } else {
-                        die($this->listenerObject->z_xlt("Unable to modify application config Please give write permission to")." $fileName");
-                    }
-
                     $status = true;
                 }
             } else {
+                // TODO: there doesn't appear to be any methodology in how to load these custom registered modules... which seems pretty odd.
+                // there aren't any in the system... but why have this then?
```

Comment:
sounds like there are several zend existential questions :)

Reply:
My only concern with removing the 'custom' module piece is if anyone has built any of those and relies on them working.  I'm going to guess that the answer is no since there doesn't appear to be a way to load them without someone really hacking the code.

Reply:
@matrix-amiel Do you use any custom_modules that are not zend modules?

Reply:
No, only zend